lib/Echidna/Output/Corpus.hs

```diff
@@ -26,3 +33,19 @@ loadTxs dir = do
   putStrLn ("Loaded " ++ show (length txSeqs) ++ " transaction sequences from " ++ dir)
   pure txSeqs
   where readCall f = decodeStrict <$> BS.readFile f
+
+-- setup a handler to save to corpus in the background while tests are running
+setupCorpusSaver :: (MonadReader Env m, MonadIO m) => m ()
+setupCorpusSaver = do
+  env <- ask
+  maybe (pure ()) (addEventHandler . saveEvent env) env.cfg.campaignConf.corpusDir
+  where
+    saveEvent env dir (_, _, event) = maybe (pure ()) (saveFile env dir) $ getEventInfo event
+
+    getEventInfo (TestFalsified test)    = Just ("reproducers", test.reproducer)
+    getEventInfo (TestOptimized test)    = Just ("reproducers", test.reproducer)
+    getEventInfo (TestSimplified test)   = Just ("reproducers", test.reproducer)
+    getEventInfo (NewCoverage _ _ _ txs) = Just ("coverage", txs)
+    getEventInfo _ = Nothing
+
+    saveFile env dir (subdir, txs) = unless (null txs) $ spawnThread env $ saveTxs (dir </> subdir) [txs]
```

Comment:
Like I mentioned earlier, if anything goes wrong while saving the file, we won't know.